File: RobWork/example/snippets/ex-frame-to-frame-transforms.cpp
#include <rw/kinematics/FKRange.hpp>
#include <rw/kinematics/State.hpp>

#include <vector>

using rw::math::Transform3D;
using namespace rw::kinematics;

std::vector<Transform3D<> > frameToFrameTransforms(
    const Frame& a,
    const Frame& b,
    const State& tree_structure,
    const std::vector<State>& states)
{
    FKRange fk(&a, &b, tree_structure);

    std::vector<Transform3D<> > result;
    for(const State& state : states) {
        result.push_back(fk.get(state));
    }
    return result;
}

File: RobWork/example/java/src/ExFrameToFrameTransforms.java
import java.util.Iterator;

import org.robwork.sdurw.*;

public class ExFrameToFrameTransforms {
    public static Transform3DdVector frameToFrameTransforms(
            Frame a, Frame b, State tree_structure, StateVector states)
    {
        FKRange fk = new FKRange(a, b, tree_structure);

        Transform3DdVector result = new Transform3DdVector();
        Iterator<State> iterator = states.iterator();
        while(iterator.hasNext()) {
            State state = iterator.next();
            result.add(fk.get(state));
        }
        return result;
    }
}